app/src/main/java/com/github/se/gomeet/ui/mainscreens/Trends.kt

```diff
@@ -159,11 +175,183 @@ fun Trends(
+      val startOfWeek = currentDate.clone() as Calendar
+      startOfWeek.set(Calendar.DAY_OF_WEEK, startOfWeek.firstDayOfWeek)
+      val endOfWeek = startOfWeek.clone() as Calendar
+      endOfWeek.add(Calendar.DAY_OF_WEEK, 6)
+
+      val eventCalendar = Calendar.getInstance().apply { time = eventDate }
+
+      val isThisWeek = eventCalendar.after(currentDate) && eventCalendar.before(endOfWeek)
+      val isToday =
+          currentDate.get(Calendar.YEAR) == eventCalendar.get(Calendar.YEAR) &&
+              currentDate.get(Calendar.DAY_OF_YEAR) == eventCalendar.get(Calendar.DAY_OF_YEAR)
+
+      val dayFormat =
+          if (isThisWeek) {
+            SimpleDateFormat("EEEE", Locale.getDefault())
+          } else {
+            SimpleDateFormat("dd/MM/yy", Locale.getDefault())
+          }
+
+      val timeFormat = SimpleDateFormat("HH:mm", Locale.getDefault())
+
+      val dayString =
+          if (isToday) {
+            "Today"
+          } else {
+            dayFormat.format(eventDate)
+          }
+      val timeString = timeFormat.format(eventDate)
+
+      Box(
+          modifier =
+              Modifier.padding(8.dp)
+                  .background(Color.Gray, shape = RoundedCornerShape(16.dp))
+                  .fillMaxSize()
+                  .clickable {
+                    nav.navigateToEventInfo(
+                        eventId = event.eventID,
+                        title = event.title,
+                        date = dayString,
+                        time = timeString,
+                        organizer = event.creator,
+                        rating = 0.0,
+                        description = event.description,
+                        loc = LatLng(event.location.latitude, event.location.longitude))
+                  }
+                  .clip(RoundedCornerShape(16.dp)),
+          contentAlignment = Alignment.Center) {
+            Image(
+                painter = painter,
+                contentDescription = null,
+                modifier = Modifier.fillMaxSize(),
+                contentScale = ContentScale.Crop)
+            Box(modifier = Modifier.align(Alignment.BottomCenter).padding(8.dp)) {
+              Box(
+                  modifier =
+                      Modifier.clip(RoundedCornerShape(8.dp))
+                          .background(Color.Black.copy(alpha = 0.5f))
+                          .padding(horizontal = 8.dp, vertical = 4.dp)) {
+                    Text(
+                        text = event.title,
+                        color = Color.White,
+                        fontSize = 17.sp,
+                    )
+                  }
+            }
+          }
+    }
+
+    HorizontalPagerIndicator(
+        pagerState = pagerState,
+        modifier = Modifier.align(Alignment.CenterHorizontally).padding(16.dp),
+        inactiveColor = MaterialTheme.colorScheme.inverseOnSurface,
+        activeColor = MaterialTheme.colorScheme.onSurface)
+  }
+}
+
+enum class SortOption {
```

Comment:
This can sound a bit stupid, but I think it would be better for refactoring to create a new file class in the `model` package just to store this enum class even if it's a very trivial enum class, you could call it `TrendsSortOption` if you want !